File: tks/RepositoriesAdapters/src/main/kotlin/mappers/EmailMapper.java
package mappers;

import exceptions.TypeValidationFailedException;
import model.values.Email;
import org.mapstruct.Mapper;

@Mapper
public interface EmailMapper {

    default String toValue(Email email){
        return email.getValue();
    }

    default Email toEmail(String value) throws TypeValidationFailedException {
        return new Email(value);
    }
}

File: tks/Common/src/main/kotlin/common/mappers/EmailMapper.kt
package common.mappers

import domain.model.values.Email
import org.mapstruct.Mapper
import org.mapstruct.Named

@Mapper
interface EmailMapper {

    @Named("toEmail")
    @JvmDefault
    fun toEmail(source: String?): Email? = if (source != null) Email(source) else null

    @Named("toEmailStr")
    @JvmDefault
    fun toString(source: Email?): String? = source?.value
}